File: src-tauri/src/models/province.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize)]
pub struct Province {
    #[serde(rename = "id")]
    pub id: i32,
    #[serde(rename = "countryId")]
    pub country_id: i32,
    #[serde(rename = "name")]
    pub name: String,
}

File: src/models/province.ts
import { Serializable } from './serializable';

export class Province implements Serializable<Province> {
  id: number;
  countryId: number;
  name: string;
  deserialize(input: any): Province {
    this.id = input.id;
    this.countryId = input.countryId;
    this.name = input.name;
    return this;
  }
}
